R/dbWorker.R

```diff
@@ -1,19 +1,37 @@
 
 
-#' Utility wrapper method for queries handling mysql and sqlite
-#' @param sql sql query to run
-#' @returns returns the query result, Note: The returned data type is dependent on query
+#' @title Get results from an SQL query to a RaMP database
+#'
+#' @description
+#'
+#' Utility function to execute the provided SQL query and get its results from
+#' a RaMP-DB database.
+#'
+#' @param sql `character(1)` with the SQL query to run.
+#'
+#' @param db [RaMP()] object representing a RaMP database. By default
+#'     (`db = RaMP()`) a connection to the most recent version is established,
+#'     which will be downloaded first if it does not already exist in the local
+#'     cache.
+#' 
+#' @return The result from the query.
+#'
+#' @importMethodsFrom DBI dbGetQuery
+#' 
 #' @export
-runQuery <- function(sql) {
-  conn = RaMP::connectToRaMP()
-  if(get("is_sqlite",pkg.globals)) {
-    rs <- RSQLite::dbGetQuery(conn, sql)
-    RSQLite::dbDisconnect(conn)
-  } else {
-    rs <- RMariaDB::dbGetQuery(conn, sql)
-    RMariaDB::dbDisconnect(conn)
-  }
-  return(rs)
+runQuery <- function(sql, db = RaMP()) {
+    con <- .dbcon(db)
+    on.exit(dbDisconnect(con))
+    dbGetQuery(con, sql)
```

Comment:
since we're importing the `dbGetQuery` from `DBI` there is no need to specifically call the `RSQLite` or `RMariaDB` specific implementations - the R dispatcher will automatically call the respective method depending on the class of the database connection.

Reply:
OK. That makes sense. That will clean things up. Thanks. It's great to clean that up.